src/sentry/utils/performance_issues/performance_detection.py

```diff
@@ -173,8 +182,15 @@ def _detect_performance_problems(data: Event, sdk_span: Any) -> List[Performance
 
 
 def prepare_problem_for_grouping(
-    problem: PerformanceSpanProblem, data: Event, detector_type: DetectorType
+    problem: Union[PerformanceProblem, PerformanceSpanProblem],
+    data: Event,
+    detector_type: DetectorType,
 ) -> PerformanceProblem:
+    # Don't transform if the caller has already done the work for us.
+    # (TBD: All detectors should get updated to just return PerformanceProblem directly)
```

Comment:
IMO anyway!

Reply:
does this mean that the new detector will store a list of `PerformanceProblem`?

Reply:
I like that pattern, I think it makes sense since each detector would have its own fingerprinting rules.